### PR TITLE
refactor: hoist Docker build context concerns + fix: missing ecosystem config

### PR DESCRIPTION
**Pull request by rhyslbw**
# Description

https://community.rosetta-api.org/t/improvement-proposal-embrace-native-remote-build-contexts-to-improve-dockerfile-design/204

I was intending to wait for feedback, but am hitting significant complexity with implementing a robust CI process to establish a merge preview context, which is awkward given it's there inherently. The original intent was to manage a dev image, but it would require two cache images to provide the CI build with a source, which in addition to the added weight, is only testing an approximation.

Also, config handling was changed in 42a1db1, removing the previous default assignments for PORT and BIND_ADDRESS that was relied up by the the Dockerfile ecosystem config. 

# Proposed Solution

- remove dev.Dockerfile and setup script
- `COPY` from context, rather than git clone
- update example build commands to demonstrate same outcome
- refactor automation to use locally managed versions,
maintained in source control, except for pre-release
- add `cache_from` to the docker-compose.yml
- Specify the missing ENVs

# Important Changes Introduced

- The suggested build commands have changed, as documented.
- There is no longer an internal version to maintain in the Dockerfile.

# Testing
This PR is covered by defined tests including
- Automated CI 
- Manual trigger of the pre-release workflow: https://github.com/input-output-hk/cardano-rosetta/actions/runs/237666935

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,10 +42,9 @@ docker run \
   --name cardano-rosetta \
   -p 8080:8080 \
   -v cardano-rosetta:/data \
-  cardano-rosetta:0.2.0 
+  cardano-rosetta:0.2.1
```
Comment (AlanVerbner) [inline README.md:45]: I think it's fine although it's super easy to forget to change it when performing a relase ... do you think it's a better way to state the version here?

Comment (rhyslbw) [reply README.md:45]: Yes agree. We can automate a find and replace easily as part of a version bump action sequence

**Comment by issoupadrink** [comment]
You've changed from `inputs.cardano-rosetta-version` to `inputs.tag` in the Dockerfiles, but the actions still use `inputs.cardano-rosetta-version`. Do they need to be changed too? 

https://github.com/input-output-hk/cardano-rosetta/blob/4ba82ad59084397772dbe31be62d84994c880469/.github/actions/build_cardano_rosetta/action.yml

https://github.com/input-output-hk/cardano-rosetta/blob/4ba82ad59084397772dbe31be62d84994c880469/.github/actions/smoke_test_cardano_rosetta/action.yml

**Review by issoupadrink — APPROVED** [review]
LGTM!

**Pull request merged**